// @wix-pilot/core/dist/common/extract/extractTaggedOutputs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.extractAutoPilotReviewOutputs = extractAutoPilotReviewOutputs;
exports.extractAutoPilotStepOutputs = extractAutoPilotStepOutputs;
const reviews_utils_1 = require("../../performers/auto-performer/reviews/reviews-utils");
const BASE_AUTOPILOT_STEP = {
    screenDescription: { tag: "SCREENDESCRIPTION", isRequired: true },
    thoughts: { tag: "THOUGHTS", isRequired: true },
    action: { tag: "ACTION", isRequired: true },
    goalSummary: { tag: "GOAL_SUMMARY", isRequired: false },
};
const AUTOPILOT_REVIEW_SECTION = {
    summary: { tag: "SUMMARY", isRequired: false },
    findings: { tag: "FINDINGS", isRequired: false },
    score: { tag: "SCORE", isRequired: false },
};
function extractTaggedOutputs({ text, outputsMapper, }) {
    const entries = Object.keys(outputsMapper).map((key) => {
        const { tag, isRequired } = outputsMapper[key];
        const regex = new RegExp(`<${tag}>(.*?)</${tag}>`, "s");
        const match = text.match(regex);
        if (match) {
            return [key, match[1].trim()];
        }
        else if (!isRequired) {
            return [key, undefined];
        }
        else {
            throw new Error(`Missing field for required tag <${tag}>`);
        }
    });
    return Object.fromEntries(entries);
}
/**
 * Extracts review outputs (summary, findings, score) from text with review tags
 */
function extractAutoPilotReviewOutputs(text) {
    return extractTaggedOutputs({
        text,
        outputsMapper: AUTOPILOT_REVIEW_SECTION,
    });
}
/**
 * Extracts autopilot step outputs including dynamic review sections
 */
function extractAutoPilotStepOutputs(text, reviewTypes = []) {
    const reviewSections = (0, reviews_utils_1.reviewConfigsToOutputsMapping)(reviewTypes);
    const outputsMapper = { ...BASE_AUTOPILOT_STEP, ...reviewSections };
    return extractTaggedOutputs({ text, outputsMapper });
}
